packages/transactions/src/create_transaction.ts

```diff
@@ -1,9 +1,9 @@
 import { PublicKey } from '@near-js/crypto';
-import BN from 'bn.js';
 
 import { Action } from './actions';
 import { Transaction } from './schema';
 
-export function createTransaction(signerId: string, publicKey: PublicKey, receiverId: string, nonce: BN | string | number, actions: Action[], blockHash: Uint8Array): Transaction {
-    return new Transaction({ signerId, publicKey, nonce: new BN(nonce), receiverId, actions, blockHash });
+export function createTransaction(signerId: string, publicKey: PublicKey, receiverId: string, nonce: bigint | string | number, actions: Action[], blockHash: Uint8Array): Transaction {
+    const bigIntNonce = typeof nonce === 'string' || typeof nonce === 'number' ? BigInt(nonce) : nonce;
+    return new Transaction({ signerId, publicKey, nonce: bigIntNonce, receiverId, actions, blockHash });
```

Comment:
`bigIntNonce` is misleading since it can be of type `bigint | string | number`
```suggestion
    const txNonce = typeof nonce === 'string' || typeof nonce === 'number' ? BigInt(nonce) : nonce;
    return new Transaction({ signerId, publicKey, nonce: txNonce, receiverId, actions, blockHash });
```

Reply:
Also I think it may be more readable and concise if you do the following
```suggestion
    const txNonce = typeof nonce === 'bigint' ? nonce :  BigInt(nonce);
    return new Transaction({ signerId, publicKey, nonce: txNonce, receiverId, actions, blockHash });
```